android/guava/src/com/google/common/cache/CacheBuilder.java

```diff
@@ -231,13 +231,16 @@ public CacheStats snapshot() {
           });
   static final CacheStats EMPTY_STATS = new CacheStats(0, 0, 0, 0, 0, 0);
 
-  static final Supplier<StatsCounter> CACHE_STATS_COUNTER =
-      new Supplier<StatsCounter>() {
-        @Override
-        public StatsCounter get() {
-          return new SimpleStatsCounter();
-        }
-      };
+  /*
+   * We avoid using a method reference here for now: Inside Google, CacheBuilder is used from the
+   * implementation of a custom ClassLoader that is sometimes used as a system classloader. That's a
+   * problem because method-reference linking tries to look up the system classloader, and it fails
+   * because there isn't one yet.
+   *
+   * I would have guessed that a lambda would produce the same problem, but maybe it's safe because
+   * the lambda implementation is generated as a method in the _same class_ as the usage?
+   */
```

Comment:
**Suggestion**: The refactoring to use a lambda expression for CACHE_STATS_COUNTER is a good improvement for readability and conciseness. However, ensure that the comment explaining the avoidance of method references is clear and accurate. Consider elaborating on the specific scenarios where method references might cause issues, based on real-world cases or theoretical concerns. <br> **Code Suggestion**: 
 ```
 +  /*
 +   * We avoid using a method reference here due to potential issues with custom ClassLoader implementations used as system classloaders. Method-reference linking might attempt to look up the system classloader prematurely, leading to failures in specific contexts.
 +   */
 ```